web/src/scripts/components/filetree/Node.js

```diff
@@ -137,27 +175,34 @@ export default class Node extends Component {
   onMouseLeave = () => this.setState({hover: false})
 
   render() {
-    const {styles, node, metadata, depth} = this.props
+    const {styles, node, metadata, depth, connectDragSource, connectDropTarget, isDraggingOver} = this.props
     const {type, name, path} = node
     const {expanded, selected} = metadata
 
     const {hover} = this.state
 
     return (
-      <div
-        style={styles.getNodeStyle(depth, selected, hover)}
-        onMouseEnter={this.onMouseEnter}
-        onMouseLeave={this.onMouseLeave}
-      >
-        {isDirectory(type) && (
-          <NodeCaret expanded={expanded}/>
-        )}
-        <div style={isDirectory(type) ? styles.folderIcon : styles.fileIcon} />
-        <div className={'flex-variable'} style={styles.nodeText}>
-          {name}
+      connectDragSource(connectDropTarget(
+        <div
+          style={styles.getNodeStyle(depth, selected, hover || isDirectory(type) && isDraggingOver)}
+          onMouseEnter={this.onMouseEnter}
+          onMouseLeave={this.onMouseLeave}
+        >
+          {isDirectory(type) && (
+            <NodeCaret expanded={expanded}/>
+          )}
+          <div style={isDirectory(type) ? styles.folderIcon : styles.fileIcon} />
+          <div className={'flex-variable'} style={styles.nodeText}>
+            {name}
+          </div>
+          {this.renderButton()}
         </div>
-        {this.renderButton()}
-      </div>
+      ))
     )
   }
 }
+
+export default flow(
```

Comment:
Can we use decorators for these? I don't think we use this `flow` pattern anywhere else